mypy/type_visitor.py

```diff
@@ -347,3 +349,121 @@ def query_types(self, types: Iterable[Type]) -> T:
                 self.seen_aliases.add(t)
             res.append(t.accept(self))
         return self.strategy(res)
+
+
+class TypeQueryBool(SyntheticTypeVisitor[bool]):
+    """Specialized visitor for boolean strategies
+
+    strategy is used to combine results for a series of types,
+    Cases involve a boolean query using `any` or `all`.
+
+    Note: this visitor keeps an internal state (tracks type aliases to avoid
+    recursion), so it should *never* be re-used for querying different types,
+    create a new visitor instance instead.
+
+    # TODO: check that we don't have existing violations of this rule.
+    """
+    STRATEGY_ANY = 0  # type: Final[int]
```

Comment:
Shouldn't it be just `: Final`?

```python
from typing_extensions import Final


class Test(object):
    a = 1  # type: Final
    b = 2  # type: Final[int]

reveal_type(Test.a)  # Revealed type is 'Literal[1]?'
reveal_type(Test.b)  # Revealed type is 'builtins.int'
```

Reply:
The doc says that if we omit the type, mypy infers it
https://mypy.readthedocs.io/en/stable/final_attrs.html#syntax-variants
I don't know what conclusion to draw from your example, is it better to have a Literal or builtins.int?

Reply:
I am not sure about the mypy team, but I personally prefer Literal, because it is more specific. And constants should be specific. 👍

Reply:
I changed them to Final 👌